src/Sarif.Sarifer/BackgroundAnalyzerBase.cs

```diff
@@ -50,68 +40,70 @@ public abstract class BackgroundAnalyzerBase : IBackgroundAnalyzer
         public abstract string ToolSemanticVersion { get; }
 
         /// <inheritdoc/>
-        public async Task AnalyzeAsync(string path, string text, CancellationToken cancellationToken)
+        public async Task<Stream> AnalyzeAsync(string path, string text, CancellationToken cancellationToken)
         {
             text = text ?? throw new ArgumentNullException(nameof(text));
 
             string solutionDirectory = await GetSolutionDirectoryAsync().ConfigureAwait(continueOnCapturedContext: false);
 
-            using (Stream stream = new MemoryStream())
-            using (TextWriter writer = new StreamWriter(stream, Encoding.UTF8))
+            // If we don't have a solutionDirectory, then, we don't need to analyze.
+            if (string.IsNullOrEmpty(solutionDirectory))
             {
-                using (SarifLogger sarifLogger = this.MakeSarifLogger(writer))
-                {
-                    sarifLogger.AnalysisStarted();
-
-                    // TODO: What do we do when path is null (text buffer with no backing file)?
-                    Uri uri = path != null ? new Uri(path, UriKind.Absolute) : null;
+                return Stream.Null;
+            }
 
-                    this.AnalyzeCore(uri, text, solutionDirectory, sarifLogger, cancellationToken);
+            var stream = new MemoryStream();
+            var writer = new StreamWriter(stream, Encoding.UTF8);
```

Comment:
>writer [](start = 16, length = 6)

It looks like we are leaking the writer. Can we do a using with the writer? Isn't there an option on the StreamWriter ctor which says "Don't dispose the stream when you dispose the writer"?